File: hangman/Hang.java
// TODO: Handle repeat characters
import java.util.Scanner;
import java.util.HashSet;

public class Hang {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    char[] word = sc.next().toCharArray();
    char[] answer = sc.next().toCharArray();
    sc.close();

    int errors = 0;
    HashSet<Character> letters = new HashSet<Character>();
    HashSet<Character> answers = new HashSet<Character>();
    for(char c : word)
      letters.add(c);

    for(char c : answer){
      if(answers.add(c)){
        if(letters.remove(c))
          if(letters.isEmpty()){
            System.out.println("WIN");
          }
        else
          if(++errors >= 10)
            System.out.println("LOSE");
      }
    }
  }
}

File: hangman/Hang.py
word = set(input())
guesses = input()

wrong_guess = 0
for letter in guesses:
    if letter in word:
        word.remove(letter)

        if len(word) == 0:
            print('WIN')
            break
    else:
        wrong_guess += 1
        if wrong_guess == 10:
            print('LOSE')
            break
